styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -190,9 +194,32 @@ public void init() {
         .watch(new PodWatcher(Integer.parseInt(resourceVersion)));
   }
 
-  private void pollPods() {
+  void examineRunningWFISandAssociatedPods(PodList podList) {
+    final Set<WorkflowInstance> activeWorkflowInstances = stateManager.activeStates().entrySet()
```

Comment:
Since RunState includes the WorkflowInstance, we can just use `.values()` followed by something like:
```
.filter(runState -> !runState.equals(RUNNING))
.map(RunState::workflowInstance) 
```